### DISCUSSION TITLE
How should the Console display $inflight constructs results from the jsify implementation

### DISCUSSION THREAD

**Discussion by ainvoner** [winglang/wing#970] [General]
In [this user defined annotation](https://github.com/winglang/wing/pull/869) PR we introduced a new auto generated $inflight "construct" that is also added to the tree.json output.
For this code:
```
bring cloud;

let bucket = new cloud.Bucket();
let queue = new cloud.Queue();

let handler = inflight (message: str): str => {
  bucket.put("hello.txt", "Hello, ${message}!");
  return message;
};

queue.on_message(handler);

let counter = new cloud.Counter(initial: 0);
new cloud.Function(inflight (message: str): str => {
  counter.inc();
  return message;
}) as "IncrementCounter";
```
we are getting new additions to the tree.json, such as:

```
      "$Inflight7b7e2066": {
        "id": "$Inflight7b7e2066",
        "path": "root/$Inflight7b7e2066",
        "attributes": {
          "wing:resource:stateful": false,
          "wing:resource:connections": []
        },
        "constructInfo": {
          "fqn": "constructs.Construct",
          "version": "10.0.130"
        }
      },
```
Which make our Console view looks like:
 
![Screenshot 2022-12-21 at 13 33 32](https://user-images.githubusercontent.com/2538825/208895890-13c9b907-6749-4a9e-9bf8-ea5afe2ff87d.png)

When [this visual annotation](https://github.com/winglang/wing/issues/962) issue will be implemented, we could add a default visibility (false) for those types of $inflight auto generated "constructs".

We need to find a way of to remove those from the Console until the above issue will be solved.

My suggestion for now would be to ignore all "$inflight" prefixed resources and constructs.

**Regarding the additional `cloud.Queue-OnMessageHandler-31b6af0d` (can be seen in the image above), I don't have an idea for how to ignore it.**

@Chriscbr @eladb @skyrpex @polamoros 
What do you say?

**Comment by skyrpex** [comment]
For now, ignoring resources based on the `$inflight` prefix sounds good to me. Then, rely on the visual annotation as you said.

Until then, looks like there's nothing else we could do to improve the experience.

**Reply by eladb** [reply]
OK to ignore explicitly for now, but I also think that adding the simple annotations as suggested in #962 will be very easy to do, so I'd consider just doing that.

**Reply by ainvoner** [reply]
Updated the issue priority to P1 and added it to our next sprint.
We need it done in order to work on several issues in our Console
